src/hotspot/share/gc/shenandoah/shenandoahMarkCompact.cpp

```diff
@@ -114,14 +116,14 @@ void ShenandoahMarkCompact::do_it(GCCause::Cause gc_cause) {
 
     // b. Cancel concurrent mark, if in progress
     if (heap->is_concurrent_mark_in_progress()) {
-      heap->concurrent_mark()->cancel();
+      ShenandoahConcurrentMark::cancel();
       heap->set_concurrent_mark_in_progress(false);
     }
     assert(!heap->is_concurrent_mark_in_progress(), "sanity");
 
     // c. Update roots if this full GC is due to evac-oom, which may carry from-space pointers in roots.
     if (has_forwarded_objects) {
-      heap->concurrent_mark()->update_roots(ShenandoahPhaseTimings::full_gc_update_roots);
+      ShenandoahConcurrentMark::update_roots(ShenandoahPhaseTimings::full_gc_update_roots);
```

Comment:
Same as with cancel() above.

Reply:
ShenandoahConcurrentMark is a temporary placement for update_roots(), it will be moved to ShenandoahGC in later refactoring,  so this call becomes instance method call, as ShenandoahMarkCompact is a subclass of ShenandoahGC.